### PR TITLE
fix!: improve message validation

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #1370, #634

- Improve the validation of IBC packet data. 
- Validate the consumer packet data received by the provider before handling the packets.
- Refactor `OnRecvPacket` keeper methods to return an error instead of an IBC ack. 

EDIT: validation of `MsgSubmitConsumerDoubleVoting` will be added in a following PR

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if state-machine breaking change (i.e., requires coordinated upgrade)
* [x] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: x/ccv/consumer/keeper/relay.go [hunk 1]
```diff
@@ -25,7 +24,12 @@ import (
 //
 // Note: CCV uses an ordered IBC channel, meaning VSC packet changes will be accumulated (and later
 // processed by ApplyCCValidatorChanges) s.t. more recent val power changes overwrite older ones.
-func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, newChanges ccv.ValidatorSetChangePacketData) exported.Acknowledgement {
+func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, newChanges ccv.ValidatorSetChangePacketData) error {
+	// validate packet data upon receiving
+	if err := newChanges.ValidateBasic(); err != nil {
```
Comment (MSalopek) [inline x/ccv/consumer/keeper/relay.go:29]: Consider calling this `Validate` and not `ValidateBasic`

By convention, `ValidateBasic` is used on Tx Message types. This is a type of packet so the naming can be confusing.

File: x/ccv/consumer/keeper/relay.go [hunk 2]
```diff
@@ -25,7 +24,12 @@ import (
 //
 // Note: CCV uses an ordered IBC channel, meaning VSC packet changes will be accumulated (and later
 // processed by ApplyCCValidatorChanges) s.t. more recent val power changes overwrite older ones.
-func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, newChanges ccv.ValidatorSetChangePacketData) exported.Acknowledgement {
```
Comment (MSalopek) [inline x/ccv/consumer/keeper/relay.go:28]: Are you sure about this? You're changing the signature of an ibc module method?

**Review by MSalopek — APPROVED** [review]
LGTM in general.

Made some nitpick comments

File: x/ccv/consumer/keeper/relay.go (hunk 2 above)
Comment (mpoke) [reply x/ccv/consumer/keeper/relay.go:28]: This is the signature of a keeper function called by an IBC module method. The IBC logic is in i`bc_module.go` and I would like as much as possible to keep it there. See https://github.com/cosmos/ibc-go/blob/9db308b8fe1713f8d503b3e54996fc326fad28be/modules/apps/transfer/ibc_module.go#L193

File: x/ccv/consumer/keeper/relay.go (hunk 1 above)
Comment (mpoke) [reply x/ccv/consumer/keeper/relay.go]: Good call. Especially as this function is not (yet) called on `checkTx`, while `ValidateBasic` usually is. See https://github.com/cosmos/interchain-security/pull/1460/commits/5488484374096426ed33c0cb76aacba3313f15b6.

File: x/ccv/types/wire.go [hunk 3]
```diff
@@ -75,13 +87,19 @@ func NewSlashPacketDataV1(validator abci.Validator, valUpdateId uint64, infracti
 	}
 }
 
-func (vdt SlashPacketData) ValidateBasic() error {
-	if len(vdt.Validator.Address) == 0 || vdt.Validator.Power == 0 {
-		return errorsmod.Wrap(ErrInvalidPacketData, "validator fields cannot be empty")
+func (vdt SlashPacketData) Validate() error {
+	// vdt.Validator.Address must be a consensus address
+	if err := sdk.VerifyAddressFormat(vdt.Validator.Address); err != nil {
```
[the remark this answers is not in the document]
Comment (mpoke) [reply x/ccv/types/wire.go]: This doesn't work :(

[the remark this answers is not in the document]
Comment (sainoe) [reply x/ccv/types/wire.go:92]: Sorry, the updated suggestion should work!

**Pull request merged**